### PR TITLE
Update docu about OData client

### PR DESCRIPTION
**Pull request by Emdee89**
## Definition of Done

Please consider all items and remove only if not applicable.

- [ ] Tests created/adjusted for your changes.
- [ ] Release notes updated.
  * Provide sufficient context so that each entry can be understood on its own.
  * Be specific about names of functions, classes, modules, etc.
  * Describe when or where this is relevant
  * Use indicative and present tense. For example, write "Provide function `name` that does X in order to Y" over "Now X can be done by calling a new function".
- [ ] PR title adheres to [conventional commit guidelines](https://www.conventionalcommits.org) (please note that only `fix:` and `feat:` will end up in the release notes)
- [ ] If applicable: Properly documented (JSDoc of public API)
- [ ] If applicable: Check if `node run doc` still works.

### CODE REVIEW

**Review by artemkovalyov — COMMENTED** [review]
I look good. Do I understand correctly it has to be updated when we actually release it?

File: docs/java/features/odata/type-safe-client-odata-v2.md [hunk 1]
```diff
@@ -27,27 +27,32 @@ The typed OData v2 client allows to build type-safe OData v2 requests for a give
...
-- Has good performance, but not optimized for optimal metadata handling
+:::tip use Tabs to see the API difference
+SAP Cloud SDK `3.29.0` releases the productive version of the new, improved OData client. Consequently, the APIs of the old OData client have been deprecated.
+:::
 
-### Improved but still experimental client version
+### The new improved client
 
-- Currently released as `Beta`, meaning the API might see breaking changes
+As of SAP Cloud SDK `3.29.0` we offer a generally available improved OData client.
```
Comment (artemkovalyov) [inline docs/java/features/odata/type-safe-client-odata-v2.md:36]:
```suggestion
As of SAP Cloud SDK `3.29.1` we offer a generally available improved OData client.
```

**Review by artemkovalyov — APPROVED** [review]
Looks good to me @Emdee89 , I suggest updating version to 3.29.1 as we'll have the only HotFix now.
Please, ping me after suggestion is committed for this to be merged.

**Comment by artemkovalyov** [comment]
Had this been commited before Frank renamed everything to MDX?

File: docs/java/features/odata/type-safe-client-odata-v2.md [hunk 2]
```diff
@@ -27,27 +27,32 @@ The typed OData v2 client allows to build type-safe OData v2 requests for a give
...
 The SDK has two versions of type-safe OData v2 client.
 
-### A stable client version
-
-- Stable and well tested
-- Based on 3rd party dependencies, which slows down release of advanced features
-- Doesn't share a code base with OData v4 client
-- Has good performance, but not optimized for optimal metadata handling
+:::tip use Tabs to see the API difference
+SAP Cloud SDK `3.29.0` releases the productive version of the new, improved OData client. Consequently, the APIs of the old OData client have been deprecated.
```
Comment (Emdee89) [inline docs/java/features/odata/type-safe-client-odata-v2.md:31]:
```suggestion
SAP Cloud SDK `3.29.1` releases the productive version of the new, improved OData client. Consequently, the APIs of the old OData client have been deprecated.
```

File: docs/java/features/odata/type-safe-client-odata-v2.md [hunk 3]
```diff
@@ -27,27 +27,32 @@ The typed OData v2 client allows to build type-safe OData v2 requests for a give
...
 - More and better implemented features
-- Better implementation of BATCH requests (currently in development and not yet available)
 
-### Which one to chose?
 
-- For production environment we still recommend a **stable client**, for testing and development you can leverage both clients.
-- To test improved client, please, follow simple steps in the [migration guide](#switch-to-improved-odata-vdm-beta)
+### The deprecated old client
+
+As of SAP Cloud SDK `3.29.0` we have deprecated the old OData client.
```
Comment (Emdee89) [inline docs/java/features/odata/type-safe-client-odata-v2.md:46]:
```suggestion
As of SAP Cloud SDK `3.29.1` we have deprecated the old OData client.
```

**Review by artemkovalyov — APPROVED** [review]
LGTM

**Pull request merged**